airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -0,0 +1,582 @@
+
+        @param tables_registry is the global context recording all tables created so far
+        @param from_table is the parent table to extract the nested stream from
+
+        The child stream processor will create a separate table to contain the unnested data.
+        """
+        result = StreamProcessor.create(
+            stream_name=child_name,
+            output_directory=parent.output_directory,
+            integration_type=parent.integration_type,
+            raw_schema=parent.raw_schema,
+            schema=parent.schema,
+            json_column_name=json_column_name,
+            properties=properties,
+            tables_registry=parent.tables_registry,
+            from_table=from_table,
+        )
+        result.parent = parent
+        result.is_nested_array = is_nested_array
+        result.json_path = parent.json_path + [child_name]
+        return result
+
+    @staticmethod
+    def create(
+        stream_name: str,
+        output_directory: str,
+        integration_type: DestinationType,
+        raw_schema: str,
+        schema: str,
+        json_column_name: str,
+        properties: Dict,
+        tables_registry: Set[str],
+        from_table: str,
+    ):
+        """
+        @param stream_name of the stream being processed
+
+        @param output_directory is the path to the directory where this processor should write the resulting SQL files (DBT models)
+        @param integration_type is the destination type of warehouse
+        @param raw_schema is the name of the staging intermediate schema where to create internal tables/views
+        @param schema is the name of the schema where to store the final tables where to store the transformed data
+
+        @param json_column_name is the name of the column in the raw data table containing the json column to transform
+        @param properties is the json schema description of this stream
+
+        @param tables_registry is the global context recording all tables created so far
+        @param from_table is the table this stream is being extracted from originally
+        """
+        return StreamProcessor(
+            stream_name,
+            output_directory,
+            integration_type,
+            raw_schema,
+            schema,
+            json_column_name,
+            properties,
+            tables_registry,
+            from_table,
+        )
+
+    def process(self) -> Dict:
+        """
+        @param from_table refers to the raw source table to use to extract data from
+        """
+        # Check properties
+        if not self.properties:
+            print(f"  Ignoring substream '{self.stream_name}' from {self.current_json_path()} because properties list is empty")
+            return
+
+        from_table = self.from_table
+        # Transformation Pipeline for this stream
+        from_table = self.write_model(self.generate_json_parsing_model(from_table), is_intermediate=True, suffix="ab1")
```

Comment:
I love the pipelining of these different tables! Makes it so much easier to follow. Great choice.